src/scripts/main.js

```diff
@@ -1,3 +1,390 @@
+  [3, 7, 11, 15],
+  [4, 8, 12, 16],
+];
+
+const cellLines = [
+  [1, 4],
+  [5, 8],
+  [9, 12],
+  [13, 16],
+];
+
+function moveUp() {
+  for (let col = 0; col <= 3; col++) {
+    for (let i = 0; i <= 3; i++) {
+      if (cells[cellRotated[col][i]] === '') {
+        for (let j = i; j <= 3; j++) {
+          if (cells[cellRotated[col][j]] !== '') {
+            cells[cellRotated[col][i]] = cells[cellRotated[col][j]];
+            cells[cellRotated[col][j]] = '';
+            break;
+          }
+        }
+      }
+    }
+  }
+}
+
+function sumUp() {
+  for (let col = 0; col <= 3; col++) {
+    for (let i = 0; i <= 3; i++) {
+      if (cells[cellRotated[col][i]] === cells[cellRotated[col][i + 1]]) {
+        cells[cellRotated[col][i + 1]] = cells[cellRotated[col][i]]
+          + cells[cellRotated[col][i + 1]];
+        cells[cellRotated[col][i]] = '';
+
+        score.textContent = Number(score.textContent)
+          + cells[cellRotated[col][i + 1]];
+      }
+    }
+  }
+  moveUp();
+}
+
+function moveDown() {
+  for (let col = 0; col <= 3; col++) {
+    for (let i = 3; i >= 0; i--) {
+      if (cells[cellRotated[col][i]] === '') {
+        for (let j = i; j >= 0; j--) {
+          if (cells[cellRotated[col][j]] !== '') {
+            cells[cellRotated[col][i]] = cells[cellRotated[col][j]];
+            cells[cellRotated[col][j]] = '';
+            break;
+          }
+        }
+      }
+    }
+  }
+}
+
+function sumDown() {
+  for (let col = 0; col <= 3; col++) {
+    for (let i = 3; i >= 0; i--) {
+      if (cells[cellRotated[col][i]] === cells[cellRotated[col][i - 1]]) {
+        cells[cellRotated[col][i - 1]] = cells[cellRotated[col][i]]
+          + cells[cellRotated[col][i - 1]];
+        cells[cellRotated[col][i]] = '';
+
+        score.textContent = Number(score.textContent)
+          + cells[cellRotated[col][i - 1]];
+      }
+    }
+  }
+  moveDown();
+}
+
+function checkLines(a, b) {
+  for (let i = a; i < b; i++) {
+    if (cells[i] === cells[i + 1]) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+function checkCol() {
+  for (let col = 0; col <= 3; col++) {
+    for (let i = 3; i >= 0; i--) {
+      if (cells[cellRotated[col][i]] === cells[cellRotated[col][i - 1]]) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function check() {
+  if (Object.values(cells).some((cell) => cell === '2048')) {
+    messageWin.classList.remove('hidden');
+  }
+
+  if (Object.values(cells).every((cell) => cell !== '')) {
+    for (let i = 0; i < cellLines.length; i++) {
+      if (checkLines(cellLines[i][0], cellLines[i][1])) {
+        return;
+      }
+    }
+
+    if (checkCol() === false) {
+      messageLose.classList.remove('hidden');
+    }
+  }
+}
+
+document.addEventListener('keydown', e => {
+  switch (e.key) {
+    case 'ArrowUp':
+      if (!messageStart.classList.contains('hidden')) {
+        break;
+      }
+      moveUp();
+      sumUp();
+      addNumber();
+      removeColors();
```

Comment:
instead of removing all classes and adding those that are required in the next function, try to modificate classes in the proper way inside `sycnc` function (optinally)